packages/realm/src/errors.ts

```diff
@@ -138,3 +140,45 @@ export class ClientResetError extends SyncError {
     };
   }
 }
+
+/**
+ * An error class that indicates that one or more object changes have been reverted by the server.
+ * This can happen when the client creates/updates objects that do not match any subscription, or performs writes on
+ * an object it didn't have permission to access.
+ */
+export class CompensatingWriteError extends SyncError {
+  /**
+   * The array of compensating writes performed by the server.
+   */
+  public infos: CompensatingWriteInfo[] = [];
+
+  /** @internal */
+  constructor(error: binding.SyncError) {
+    super(error);
+    if (error.compensatingWritesInfo) {
+      error.compensatingWritesInfo.forEach((element) => {
+        this.infos.push({ objectName: element.objectName, reason: element.reason, primaryKey: element.primaryKey });
+      });
+    }
+  }
+}
+
+/**
+ * The details of a compensating write performed by the server.
+ */
+export type CompensatingWriteInfo = {
+  /**
+   * The type of the object affected by the compensating write.
```

Comment:
Is the following a more accurate description?
```suggestion
   * The type of the object that caused the compensating write.
```